### ISSUE TITLE
A great improvement has been made for master branch (LJSpeech)

### ISSUE THREAD

**Issue by begeekmyfriend** [keithito/tacotron#198]
Hi all, I have a good news that we might not be worried about the slow alignment and huge quantity of dataset anymore for this project for I have found a solution to improve it. Several days ago I was still talking how many wav clips or how much duration it is needed for training.  But now I have opened on my fork [here](https://github.com/begeekmyfriend/tacotron/tree/location-sensitive-attention). I have used the default `LJSpeech` for traing and as we can see, the alignment has been quick to be learnt in 7K steps and the quantity of the dataset was less than 20 hours (less than 10K clips).
![step-7000-align](https://user-images.githubusercontent.com/6031938/44246334-51c76a80-a210-11e8-98e6-18a2d3ce8775.png)

The major modification is on the location sensitive attention model inspired by @Rayhane-mamah at commit https://github.com/begeekmyfriend/tacotron/commit/b02eee7b86019dfefb1ba36a4e193b186307daa6. I have abandoned the original `AttentionWrapper` API and the quality of audio still kept good on G&L vocoder. By the way, `AttentionWrapper` really sucks that we'd better not use it.

The Chinese mandarin version is also opened on [this](https://github.com/begeekmyfriend/tacotron) branch. Welcome to have a try.

**Comment by begeekmyfriend** [comment]
The alignment would appear in 10K steps unless your dataset is abnormal.

**Comment by jiamingkong** [comment]
![step-10000-align](https://user-images.githubusercontent.com/2761215/44627634-b480da80-a963-11e8-938a-e812e3613c56.png)

Oh wow, it did show up!

**Comment by begeekmyfriend** [comment]
You can even reduce you quantity to 1~5 hours, just try it.

**Comment by begeekmyfriend** [comment]
@keithito The modification is very large but effective. There are two major aspects. First I have replaced `AttentionWrapper` interface. Second I have added stop token target for training to learn when to stop decoding. If I shoot a PR, would you like to receive it?

**Comment by keithito** [comment]
@begeekmyfriend Yes, if you send over a PR, I would be happy to review and merge it.

**Comment by peter05010402** [comment]
hello  @begeekmyfriend ,Does the Chinese mandarin version fit thchs30 dataset?

**Comment by begeekmyfriend** [comment]
@peter05010402 Of course it does.

**Comment by peter05010402** [comment]
@begeekmyfriend thank you for reply.
With single person's wav and ten, I train 30k steps, but no align 
Could you help?

![step-30000-align](https://user-images.githubusercontent.com/42974031/45250722-e5054300-b36b-11e8-9639-945f844710bd.png)

**Comment by begeekmyfriend** [comment]
The plot looks not my fork, did you use https://github.com/begeekmyfriend/tacotron/tree/master ?

**Comment by peter05010402** [comment]
@begeekmyfriend  thank you for your reply!
Do you mean I should clone commit f8de0d7 ?

**Comment by peter05010402** [comment]
@begeekmyfriend  yes.    I want to train Chines mandarin 
what I clone is that:https://github.com/begeekmyfriend/tacotron/tree/f8de0d72d924d29578ec8402a07f3a8e730c1a8f
which is imporved by you, I guess it fit Chinese Mandarin.

**Comment by peter05010402** [comment]
@begeekmyfriend Thank you for your work.it alligned at 7k steps in thchs30!
![step-7000-align](https://user-images.githubusercontent.com/42974031/45281736-4adc0100-b50b-11e8-8106-b89b75bfe5fa.png)

but it not allign at 7k steps in one preson's speech of chths30.(The  quantity is 2.15 hours)
![step-7000-align](https://user-images.githubusercontent.com/42974031/45281954-1452b600-b50c-11e8-8d1f-1ed345f15a85.png)

**Comment by hyzhan** [comment]
@begeekmyfriend @keithito Actually not so complicated...Just like this:
from models.attention import LocationSensitiveAttention
attention_mechanism = LocationSensitiveAttention(hp.attention_dim, encoder_outputs, hparams=hp,
mask_encoder=hp.mask_encoder, memory_sequence_length=input_lengths, smoothing=hp.smoothing,
cumulate_weights=hp.cumulative_weights)
and replace origin code in AttentionWrapper "BahdanauAttention(hp.attention_depth, encoder_outputs)" with "attention_mechanism"

This attention could solve the problem of skip and repeat phenomenon.

**Comment by hccho2** [comment]
Is begeekmyfriend's model changed from the original model?
![begeekmyfriend](https://user-images.githubusercontent.com/26861167/47275695-6ea36400-d5ec-11e8-8923-2ac1a30ace69.png)

**Comment by hccho2** [comment]
Is begeekmyfriend's model changed from the original model?
![begeekmyfriend](https://user-images.githubusercontent.com/26861167/47275735-c80b9300-d5ec-11e8-9f53-e29b4e8c5d8f.png)

**Comment by hccho2** [comment]
@begeekmyfriend  Is begeekmyfriend's model changed from the original model?

![begeekmyfriend](https://user-images.githubusercontent.com/26861167/47276653-9518cd80-d5f3-11e8-978c-cba5c1d32496.png)

**Comment by hccho2** [comment]
@begeekmyfriend.     Is begeekmyfriend's model changed from the original model?
![begeekmyfriend](https://user-images.githubusercontent.com/26861167/47276704-fe004580-d5f3-11e8-8118-8612e32e8150.png)

**Comment by hccho2** [comment]
test

**Comment by sjtilney** [comment]
Thanks @begeekmyfriend I've tested your repo with the LJspeech dataset, training on a GPU, and left all hparams at their default. Yes, I did see that alignment happens more quickly, but the sound quality does not appear to be as clear compared to the mimic2 fork. I've reached 71,000 steps so far, and while the voice is clear, there is a lot of static in the background. Sample output: https://drive.google.com/file/d/1khDaXrfRhFh15JRNT87NI1Hv8FcBuFXC/view?usp=sharing

Here is the same phrase, on the mimic2 repo at 68,000 steps: https://drive.google.com/open?id=1W87m942sfuEh4UCD6n6tQXr6JSAytXRt

Any idea why there is so much static? Will more training solve this? or should I adjust some of the hparams?

Alignment using your repo after 71,000 steps
![step-71000-align](https://user-images.githubusercontent.com/9960124/49605411-432ecb80-f945-11e8-82f5-74a09060ffa5.png)

**Comment by begeekmyfriend** [comment]
@sjtilney In fact I have added some wired solution in the `mandarin` branch which have not merged into the `master` branch. I will do that soon. If you want listen the latest evaluation from G&L synthesizer. Here is the Chinese mandarin sample https://github.com/Rayhane-mamah/Tacotron-2/issues/292#issuecomment-444823633

**Comment by sjtilney** [comment]
If I'm training on the LJspeech dataset, should I use the latest commit from the Master branch?

**Comment by begeekmyfriend** [comment]
I have been traing on Chinese mandarin, you might merge the latest version of that branch into the master one.

**Comment by bjtommychen** [comment]
@begeekmyfriend 
I am confused by the hparameter setting, please help to review.
my training data is **thchs30**, sample rate is 16khz.

in https://github.com/begeekmyfriend/tacotron/tree/mandarin
the hparams.py show:
  num_mels=80,
  num_freq=2049,
  sample_rate=48000,

but I used to use.
  num_mels=80,
  num_freq=1025,
  sample_rate=16000,

Thanks.

**Comment by begeekmyfriend** [comment]
@bjtommychen http://www.data-baker.com/open_source.html

**Comment by bjtommychen** [comment]
@begeekmyfriend 
Thanks for your quick response.  I got biaobei.

I have another questions:
1. with the latest code, after training 7k steps on biaobei.  loss  is 1.11, and after  30k steps, loss is 0.83, is it correct ? what's the target loss can reach ? 
2. Is it OK to add punctuation mark in label. like this 'ni2 hao3 , you3 piao4 ma ?', 
3. biaobei is 48khz. Is it possible to make training faster after convert 48khz into 16khz. 

Thanks.

**Comment by begeekmyfriend** [comment]
1. Loss does not matter, the final evaluation does.
2. You need to add those marks on your own sentence by sentence.
3. Yes, lower sample rate can be trained faster.

**Comment by MonkeyBLuffy** [comment]
您好，我尝试使用了您的mandarin来实现中文语音的合成，我使用的数据集是清华的data_thchs30，但是我合成的结果显示出来是这个样子的：file:///home/buzhuolun/%E4%B8%8B%E8%BD%BD/datafinal/step-104000-align.png

**Comment by MonkeyBLuffy** [comment]
您好，我尝试使用了您的mandarin分支进行中文语音的合成，我使用的是清华的data_thchs30数据集，未更改参数，但是合成出来的效果确实这个样子的：
![step-104000-align](https://user-images.githubusercontent.com/48714298/76192211-3e2d8b80-621c-11ea-992a-e28f656f2c53.png)
您看问题出在那里了呢？谢谢您

**Comment by begeekmyfriend** [comment]
Please use Biaobei(标贝) open corpus.

**Comment by hishammadcor** [comment]
Hi, I have a problem with the alignment curve, at step 22000 there is no alignment at all. I am using an Arabic modified version of Keith's model on Egyptian Arabic. My dataset is 1145 sentences, 48k sample rate, and recorded under the supervision of a professional studio. I modified the hyperparameters, these are what I am using.

adam_beta1: 0.9
adam_beta2: 0.999
attention_depth: 256
batch_size: 8
cleaners: arabic_cleaners
decay_learning_rate: True
decoder_depth: 256
embed_depth: 256
encoder_depth: 256
frame_length_ms: 50
frame_shift_ms: 12.5
griffin_lim_iters: 60
initial_learning_rate: 0.001
max_frame_num: 1000
max_iters: 400
min_level_db: -100
num_freq: 2049
num_mels: 80
outputs_per_step: 1
postnet_depth: 256
power: 1.5
preemphasis: 0.97
prenet_depths: [256, 128]
ref_level_db: 20
sample_rate: 48000
Loaded metadata for 1145 examples (0.90 hours)

the data set is small, but with nearly the same size, I trained a Modern Standard Arabic model with Keith's hyperparameters and it gives good sound.

![step-22000-align](https://user-images.githubusercontent.com/32823502/141386659-4df51d8f-b051-45f3-afe3-5633d761b586.png)

could somebody help with this issue?
Thanks in advance.
